receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -234,9 +234,23 @@ func addGaugeDatapoints(rm pmetric.ResourceMetrics, ls labels.Labels, ts writev2
 	// In OTel name+type+unit is the unique identifier of a metric and we should not create
 	// a new metric if it already exists.
 
-	// TODO: Check if Scope is already present by comparing labels "otel_scope_name" and "otel_scope_version"
-	// with Scope.Name and Scope.Version. If it is present, we should append to the existing Scope.
-	m := rm.ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetEmptyGauge()
+	scopeName := ls.Get("otel_scope_name")
+	scopeVersion := ls.Get("otel_scope_version")
+	// Check if the name and version present in the labels are already present in the ResourceMetrics.
+	// If it is not present, we should create a new ScopeMetrics.
+	// Otherwise, we should append to the existing ScopeMetrics.
+	for j := 0; j < rm.ScopeMetrics().Len(); j++ {
+		scope := rm.ScopeMetrics().At(j)
+		if scopeName != "" && scopeVersion != "" && scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
```

Comment:
If scopeName or scopeVersion is empty, we can still check to see if it matches the scope, right?  Otherwise, if metrics are all missing those labels we will have many different scopes with an empty name and version.

Reply:
What I was trying to say in my comment is the case where `scopeName` and `scopeVersion` come as empty. In this scenario, I was unsure whether I should add the `dataPoints` or do nothing.

If I understood your comment correctly, my initial implementation:

```Golang
if scopeName == scope.Scope().Name() && scopeVersion == scope.Scope().Version() {
	addDatapoints(scope.Metrics().AppendEmpty().SetEmptyGauge().DataPoints(), ls, ts)
	return
}
```

covers the scope of the PR. Or did I misunderstand?

Reply:
I've just re-read the specification, and looks like we need to use the collector build information as the scope 😬 

https://opentelemetry.io/docs/specs/otel/compatibility/prometheus_and_openmetrics/#:~:text=Metrics%20which%20do%20not%20have%20an%20otel_scope_name%20or%20otel_scope_version%20label%20MUST%20be%20assigned%20an%20instrumentation%20scope%20identifying%20the%20entity%20performing%20the%20translation%20from%20Prometheus%20to%20OpenTelemetry%20(e.g.%20the%20collector%E2%80%99s%20prometheus%20receiver).

Reply:
> I'm not sure where exactly we get that information. So I'm fine doing this on a follow-up PR or in this PR, whatever is best for you @perebaj

I think it's better to leave it for the next PR. this one is frying all my brain

Reply:
hahaha fair! Do you wanna remove the draft mode?